### PR TITLE
fix: More robust STREAM_RST logic

### PR DESCRIPTION
**Pull request by emkornfield**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ x ] Ensure the tests and linter pass
- [ x ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

This change makes error matching against stream rst more robust.  Customers have started to see other variants with RST stream.  The capitalization of "Rst Stream" isn't clear, other repose have it lower cased: (e.g. https://github.com/googleapis/java-bigtable-hbase/pull/3000/files)

### CODE REVIEW

**Comment by emkornfield** [comment]
CC @kmjung @davidrabinowitz

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+   * Returns true iff the Status indicates and internal error that is retryable.
+   *
+   * <p>Generally, internal errors are not considered retryable, however there are certain transient
+   * network issues that appear as internal but are in fact retryable.
+   */
+  public static boolean isRetryableInternalStatus(Status status) {
+    return status.getCode() == Status.Code.INTERNAL
+        && status.getDescription() != null
+        && (status.getDescription().contains("Received unexpected EOS on DATA frame from server")
+            || status.getDescription().contains("Received Rst ")
```
Comment (kmjung) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java:34]: Nit: is there a reason to drop " Stream" here?

**Review by kmjung — APPROVED** [review]

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java (hunk 1 above)
Comment (emkornfield) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java:34]: Yes, from PR description:

> The capitalization of "Rst Stream" isn't clear, other repose have it lower cased: (e.g. https://github.com/googleapis/java-bigtable-hbase/pull/3000/files)

**Comment by stephaniewang526** [comment]
Please update initial commit and subject line of this PR to abide by [conventional commit rules](https://www.conventionalcommits.org/en/v1.0.0/).

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (stephaniewang526) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java:2]:
```suggestion
 * Copyright 2021 Google LLC
```

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+   *
+   * <p>Generally, internal errors are not considered retryable, however there are certain transient
+   * network issues that appear as internal but are in fact retryable.
+   */
+  public static boolean isRetryableInternalStatus(Status status) {
+    return status.getCode() == Status.Code.INTERNAL
+        && status.getDescription() != null
+        && (status.getDescription().contains("Received unexpected EOS on DATA frame from server")
+            || status.getDescription().contains("Received Rst ")
+            || status.getDescription().contains("RST_STREAM"));
```
Comment (stephaniewang526) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java:35]: Can we just check for `status.getDescription.toLowercase().contains("rst")`?

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/util/ErrorsTest.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2020 Google LLC
```
Comment (stephaniewang526) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/util/ErrorsTest.java:2]:
```suggestion
 * Copyright 2021 Google LLC
```

**Review by stephaniewang526 — CHANGES_REQUESTED** [review]

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java (hunk 3 above)
Comment (emkornfield) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java]: I don't think so.  "rst" can occur naturally in other settings.  I shortened the top one to just be " Rst ".  I also added some other messages from spark-bigquery-connector.

Comment (stephaniewang526) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/util/Errors.java]: okay lgtm

**Review by stephaniewang526 — APPROVED** [review]

**Pull request merged**